dias/analyzers/source_spectra_analyzer.py

```diff
@@ -0,0 +1,784 @@
+                        data.freq, pol=pols[ii][0], dec=src_dec, sigma=True
+                    )
+
+                flag = counter > 0.0
+
+                fitter = cal_utils.FitGaussAmpPolyPhase(poly_deg_phi=self.poly_deg_phi)
+                fitter.fit(ha, vis.real, np.sqrt(var), width=fwhm, absolute_sigma=True)
+                resid = vis.real - fitter.predict(ha)
+                resid_rms = np.std(resid, axis=-1)
+
+                if self.peak_type == "max_amp":
+                    # To have one slice in time
+                    ha_max = np.nanmedian(fitter.peak())
+                elif self.peak_type == "zero_ha":
+                    ha_max = 0.0
+                else:
+                    raise ValueError("Peak type not recognized")
+
+                peak_flux = fitter.predict(ha_max)
+
+                N2_flux = np.zeros((len(self.N2_freq), npol))
+                N2_freq_ind = _find_freq(data.freq, self.N2_freq)
+                N2_flux = peak_flux[N2_freq_ind].real
+
+                for ii, freq_export in enumerate(data.freq[N2_freq_ind]):
+                    for jj in range(npol):
+                        self.flux_metric.labels(
+                            frequency=freq_export, source=src, pol=jj
+                        ).set(N2_flux[ii, jj])
+
+                # Write to file
+                output_file = os.path.join(
+                    self.write_dir,
+                    "%s_csd_%d_%s.h5" % (src.lower(), csd, self.output_suffix),
+                )
+                self.logger.info("Writing output files...")
+
+                self.update_data_index(
+                    data.time[0], data.time[-1], filename=output_file
+                )
+
+                with h5py.File(output_file, "w") as handler:
+
+                    index_map = handler.create_group("index_map")
+                    index_map.create_dataset("freq", data=data.index_map["freq"][:])
+                    index_map.create_dataset("pol", data=np.string_(pols))
+                    index_map.create_dataset("time", data=data.time)
+                    index_map.create_dataset("ra", data=ra)
+                    index_map.create_dataset("ha", data=ha)
+
+                    dset = handler.create_dataset("vis", data=vis)
+                    dset.attrs["axis"] = np.array(["freq", "pol", "ha"], dtype="S")
+
+                    dset = handler.create_dataset("peak_vis", data=peak_flux)
+                    dset.attrs["axis"] = np.array(["freq", "pol"], dtype="S")
+
+                    dset = handler.create_dataset(
+                        "weight", data=tools.invert_no_zero(var)
+                    )
+                    dset.attrs["axis"] = np.array(["freq", "pol", "ha"], dtype="S")
+
+                    dset = handler.create_dataset("count", data=counter.astype(np.int))
+                    dset.attrs["axis"] = np.array(["freq", "pol", "ha"], dtype="S")
+
+                    index_map.create_dataset("param", data="model_fitting")
```

Comment:
Maybe someone more experienced with hdf5 can comment on this. I think the index map should have all values the model fitting parameter took, so the data indexes in the `parameter_cov` dataset can be translated to those.
@ssiegelx